src/cpp/multisolver_interface/SolverFactory.cpp

```diff
@@ -69,41 +71,47 @@ SolverAbstract::Ptr SolverFactory::create_solver(
 
 SolverAbstract::Ptr SolverFactory::create_solver(
     const std::string &solver_name) const {
-  if (solver_name == "") {
+  SolverAbstract::Ptr ret;
+  if (solver_name.empty()) {
     throw InvalidSolverNameException(solver_name, LOGLOCATION);
   } else if (isXpress_available_ && solver_name == XPRESS_STR) {
-    return std::make_shared<SolverXpress>();
+    ret = std::make_shared<SolverXpress>();
   }
 #ifdef COIN_OR
   else if (solver_name == CLP_STR) {
-    return std::make_shared<SolverClp>();
+    ret = std::make_shared<SolverClp>();
   } else if (solver_name == CBC_STR) {
-    return std::make_shared<SolverCbc>();
+    ret = std::make_shared<SolverCbc>();
   }
 #endif
   else {
     throw InvalidSolverNameException(solver_name, LOGLOCATION);
   }
+  ret->init();
+  return ret;
 }
 
 SolverAbstract::Ptr SolverFactory::create_solver(
     const std::string &solver_name, SolverLogManager &log_manager) const {
-  if (solver_name == "") {
+  if (solver_name.empty()) {
     throw InvalidSolverNameException(solver_name, LOGLOCATION);
   }
+  SolverAbstract::Ptr ret;
   if (isXpress_available_ && solver_name == XPRESS_STR) {
-    return std::make_shared<SolverXpress>(log_manager);
+    ret = std::make_shared<SolverXpress>(log_manager);
   }
 #ifdef COIN_OR
   else if (solver_name == CLP_STR) {
-    return std::make_shared<SolverClp>(log_manager);
+    ret = std::make_shared<SolverClp>(log_manager);
   } else if (solver_name == CBC_STR) {
-    return std::make_shared<SolverCbc>(log_manager);
+    ret = std::make_shared<SolverCbc>(log_manager);
   }
 #endif
   else {
     throw InvalidSolverNameException(solver_name, LOGLOCATION);
   }
+  ret->init();
+  return ret;
```

Comment:
Slight change of behavior here. `init` wasn't called at all before, but now we call it. Did you forget to remove a few calls at higher levels ?

Reply:
I've now removed all call to init at higher level (after calls to factory) reducing temporal coupling